### PR TITLE
Add model analysis `.Rmd` template

### PR DESCRIPTION
**Pull request by juliasilge**
This PR adds an `.Rmd` template to the tidymodels metapackage to demo an example model analysis. The use case here is that people choose it from the RStudio menu when they are starting a modeling analysis and it helps folks get started with their own problem.

As of now, consider this a sort of straw man for us to discuss and edit. Some questions I have:

- I included an example with no tuning. Thoughts?
- Same for a recipe; I chose to keep it simpler and link out to learn more
- What do we think about the level of narrative? Too much? Too little?
- Anything that you would want to include or take out?

### CODE REVIEW

File: inst/rmarkdown/templates/model-analysis/skeleton/skeleton.Rmd [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+```{r}
+final_fitted <- last_fit(glm_wf, penguin_split)
+collect_metrics(final_fitted)  ## metrics evaluated on the *testing* data
+```
+
+This object contains a fitted workflow that we can use for prediction.
+
+```{r}
+final_wf <- final_fitted$.workflow[[1]]
```
Comment (hfrick) [inline inst/rmarkdown/templates/model-analysis/skeleton/skeleton.Rmd:138]: do we have a function to extract that?

Comment (juliasilge) [reply inst/rmarkdown/templates/model-analysis/skeleton/skeleton.Rmd:138]: We don't, but gosh, a little helper there would be nice!

File: inst/rmarkdown/templates/model-analysis/skeleton/skeleton.Rmd [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  ggplot(aes(bill_depth_mm, bill_length_mm, color = sex, size = body_mass_g)) +
+  geom_point(alpha = 0.5) +
+  facet_wrap(~species) +
+  theme_bw()
+```
+
+
+## Build models
+
+Let's consider how to spend our data budget:
```
Comment (topepo) [inline inst/rmarkdown/templates/model-analysis/skeleton/skeleton.Rmd:41]:
```suggestion
Let's consider how to [spend our data budget](https://www.tmwr.org/splitting.html):
```

File: inst/rmarkdown/templates/model-analysis/skeleton/skeleton.Rmd [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+penguin_split <- initial_split(penguins, strata = sex)
+penguin_train <- training(penguin_split)
+penguin_test <- testing(penguin_split)
+
+set.seed(234)
+penguin_folds <- vfold_cv(penguin_train, strata = sex)
+penguin_folds
+```
+
+Let's create a **model specification** for each model we want to try:
```
Comment (topepo) [inline inst/rmarkdown/templates/model-analysis/skeleton/skeleton.Rmd:57]:
```suggestion
Let's create a [**model specification**](https://www.tmwr.org/models.html) for each model we want to try:
```

File: inst/rmarkdown/templates/model-analysis/skeleton/skeleton.Rmd [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+ranger_spec <-
+  rand_forest(trees = 1e3) %>%
+  set_engine("ranger") %>%
+  set_mode("classification")
+```
+
+To set up your modeling code, consider using the [parsnip addin](https://parsnip.tidymodels.org/reference/parsnip_addin.html) or the [usemodels](https://usemodels.tidymodels.org/) package.
+
+Now let's build a **model workflow** combining each model specification with a data preprocessor:
```
Comment (topepo) [inline inst/rmarkdown/templates/model-analysis/skeleton/skeleton.Rmd:72]:
```suggestion
Now let's build a [**model workflow**](https://www.tmwr.org/workflows.html) combining each model specification with a data preprocessor:
```

File: inst/rmarkdown/templates/model-analysis/skeleton/skeleton.Rmd [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+)
+
+ranger_rs <- fit_resamples(
+  ranger_wf,
+  resamples = penguin_folds,
+  control = control_resamples(save_pred = TRUE)
+)
+```
+
+To fit and evaluate _many_ modeling approaches together, consider using [workflowsets](https://workflowsets.tidymodels.org/). How did these two models compare?
```
Comment (topepo) [inline inst/rmarkdown/templates/model-analysis/skeleton/skeleton.Rmd:107]: If we are not going to use a workflow set, should it be mentioned? The reader might expect more details after seeing this.

Comment (juliasilge) [reply inst/rmarkdown/templates/model-analysis/skeleton/skeleton.Rmd]: Yeah, you're right -- removed 🚫

File: inst/rmarkdown/templates/model-analysis/skeleton/skeleton.Rmd (hunk 1 above)
Comment (juliasilge) [reply inst/rmarkdown/templates/model-analysis/skeleton/skeleton.Rmd]: Now we have them! 🎉

**Comment by juliasilge** [comment]
I changed the `.Rmd` template to use the new `extract_workflow()` function, and bumped the required workflows version to devel. 👍

**Pull request merged**